wg-node-lifecycle/charter.md

```diff
@@ -0,0 +1,163 @@
+- Infrastructure provisioning, deprovisioning solution or physical infrastructure lifecycle
+  management solution.
+
+## Stakeholders
+
+- SIG Apps
+- SIG Architecture
+- SIG Autoscaling
+- SIG CLI
+- SIG Cloud Provider
+- SIG Cluster Lifecycle
+- SIG Network
+- SIG Node
+- SIG Scheduling
+- SIG Storage
+
+Stakeholders span from multiple SIGs to a broad set of end users,
+public and private cloud providers, Kubernetes distribution providers,
+and cloud provider end-users. Here are some user stories:
+
+- As a cluster admin I want to have a simple interface to initiate a node drain/maintenance without
+  any required manual interventions. I also want to be able to observe the node drain via the API
+  and check on its progress. I also want to be able to discover workloads that are blocking the node
+  drain.
+- To support the new features, node maintenance, scheduler, descheduler, pod autoscaling, kubelet,
+  and other actors want to use a new eviction API to gracefully remove pods. This would enable new
+  migration strategies that prefer to surge (upscale) pods first rather than downscale them. It
+  would also allow other users/components to monitor pods that are gracefully removed/terminated
+  and provide better behaviour in terms of de/scheduling, scaling and availability.
+- As a cluster admin, I want to be able to perform arbitrary actions after the node drain is
+  complete, such as resetting GPU drivers, resetting NICs, performing software updates or shutting
+  down the machine.
+- As an end user, I would like more alternatives to blue-green upgrades, especially with special
+  hardware accelerators; it's far too expensive. I would like to choose a strategy on how to
+  coordinate the node drain and the upgrade to achieve better cost-effectiveness.
+- As a cloud provider, I need to perform regular maintenance on the hardware in my fleet. Enhancing
+  Kubernetes to help CSPs safely remove hardware will reduce operational costs.
+- Modelling the cost of doing accelerator maintenance in today's world can be massive. And since
+  hardware accelerators tend to need more love and care, having software support to coordinate
+  maintenance will reduce operational costs.
+- As a cluster admin, I would like to use a mixture of on-demand and temporary spot instances in my
+  clusters to reduce cloud expenditure. Having more reliable lifecycle and drain mechanisms for
+  nodes will improve cluster stability in scenarios where instances may be terminated by the cloud
+  provider due to cost-related thresholds.
+- As a user, I want to prevent any disruption to my pet or expensive workloads (VMs, ML with
+  accelerators) and either prevent termination altogether or have a reliable migration path. 
+  Features like `terminationGracePeriodSeconds` are not sufficient as the termination/migration can
+  take hours if not days.
+- As a user, I want my application to finish all network and storage operations before terminating a
```

Comment:
This particular use-case seems identical to the previous one. So I'd just drop it.

Reply:
There is a small distinction here. This point also applies to cheap/cattle applications. I think it is useful to keep it separate as we have a couple of issues/proposals for this.